alphalens/performance.py

```diff
@@ -247,7 +247,11 @@ def quantize_factor(factor, quantiles=5, by_group=False):
     """
 
     def quantile_calc(x, quantiles):
-        return pd.qcut(x, quantiles, labels=False) + 1
+        try:
+            return pd.qcut(x, quantiles, labels=False) + 1
+        except ValueError:
+            # This is supposed to fix "ValueError: Bin edges must be unique"
+            return pd.qcut(x.rank(method='first'), quantiles, labels=False) + 1
```

Comment:
Thanks the useful discussion.

I agree we should not merge this. Instead, I think we should catch the exception and raise a new one that says that not enough unique values are provided for the required number of quantiles and that the number of quantiles should be reduced.